### ISSUE TITLE
Virtualize list

### ISSUE THREAD

**Issue by phuoc-ng** [react-pdf-viewer/react-pdf-viewer#358]
Currently, each page is represented a separated `div` element. It isn't rendered by default, and will be rendered when the page is visible.

But this optimization is not enough for a document that has big number of pages (5000 pages, for example).
Scrolling between pages could slow down the app.

**Comment by sirkrisp** [comment]
Hi, could you prioritise this issue. It already makes a huge difference starting from 10 pages (remember each page can contain a bunch of elements). The PDF viewer is barely usable with a 800 page document (zooming out/in is extremely slow, high cpu usage etc.). 

Also I think it is not too much effort to implement considering how important this issue is. Would be great if it was as fast as the PDF.js demo from here: https://mozilla.github.io/pdf.js/web/viewer.html

Basically, only +/-5 pages (should be a parameter) of the current page should be rendered. **Currently, once a page is visible and has been rendered, it always stays rendered**. A page outside this range should be only rendered as "loading banner page" (like in the PDF.js viewer sample above - if you scroll down to the end of the document you see that the first few pages are not rendered anymore).

Thanks!

Best, Kris

**Comment by phuoc-ng** [comment]
@okilled, @handofthecode, @sirkrisp
This was implemented in the latest code. As you can see, there are lot of PRs for implementing and testing this big improvement.
It will be available in the next v3.1.0.

Here is the demo of pages which are rendered dynamically as you scroll up/down (this is recorded when testing with a 900+ pages document):

https://user-images.githubusercontent.com/57786711/149620236-a1979e4e-02f0-4afd-a81e-ab6ec11445bb.mov

**Comment by phuoc-ng** [comment]
@okilled, @handofthecode, @sirkrisp
[v3.1.0](https://react-pdf-viewer.dev/updates/) is released. Please upgrade!
